pallets/eth-bridge/src/lib.rs

```diff
@@ -794,48 +806,62 @@ pub mod pallet {
     fn process_ethereum_events_partition<T: Config>(
         active_range: &ActiveEthRange,
         partition: &EthereumEventsPartition,
-    ) {
+    ) -> Result<(), DispatchError> {
```

Comment:
This function should be infallible.
Once a voting round is concluded, the processing of each event should not affect the processing of the batch.

The errors should be handled, the appropriate events emited and processing continues.
Otherwise, the chain will be stuck in a loop of voting for the last vote and rejecting the processing.